lua/entities/magicbutton.lua

```diff
@@ -305,57 +325,321 @@ local function MagicOutcomeBountyAndPrize(ply)
-        KLEINER_OVERRIDE_TARGET = nil
+    classes["sent_ball"] = {
+        "Bouncy Ball", 10, function(ent)
+            ent:SetBallSize(24)
+            ent:GetPhysicsObject():SetVelocity(VectorRand() * 20)
+        end
+    }
+
+    classes["npc_headcrab"] = {
+        "Headcrab", math.random(1, 5), function(ent, button)
+            local pos = table.Random(navmesh.Find(button:GetPos() + button:GetUp() * 50, 100, 2000, 2000)):GetRandomPoint()
+            ent:SetPos(pos)
+        end
+    }
+
+    classes["npc_grenade_frag"] = {
+        "Live Grenade", math.random(1, 10), function(ent, button)
+            ent:GetPhysicsObject():SetVelocity(VectorRand() * 100)
+            ent:Fire("SetTimer", math.random(2, 6))
+        end
+    }
+
+    classes["dodgeball"] = {
+        "Dodgeball", 1, function(ent, button)
+            ent:GetPhysicsObject():SetVelocity(button:GetUp() * 500)
+        end
+    }
+
+    local dat, class = table.Random(classes)
+    local number = dat[2] or 1
+    local name = dat[1]
+    local func = dat[3]
+
+    for i = 1, number do
+        local ent = ents.Create(class)
+        ent:Spawn()
+        ent:SetPos(button:GetPos() + button:GetUp() * ent:BoundingRadius())
+        SafeRemoveEntityDelayed(ent, 60 * 5) --whatever we spawn, make sure it's gone after 5 minutes
+
+        if (func) then
+            func(ent, button)
+        end
+    end
+
+    local what = name or ent.PrintName or ent:GetClass()
+
+    if (number > 1) then
+        what = number .. " " .. what .. "s"
+    else
+        what = "a " .. what
+    end
+
+    return what
+end
+
+local function MagicOutcomeOverlay(ply, button)
+    local overlays = {"models/shadertest/shader4", "models/props_c17/fisheyelens", "effects/combine_binocoverlay", "models/props_combine/stasisshield_sheet", "models/shadertest/shader5", "effects/water_warp01", "effects/distortion_normal001", "effects/tp_eyefx/tpeye"}
+
+    ply:ConCommand("pp_mat_overlay_refractamount -0.06")
+    ply:ConCommand("pp_mat_overlay " .. table.Random(overlays))
+
+    timer.Simple(30, function()
+        if (IsValid(ply)) then
+            ply:ConCommand("pp_mat_overlay ''")
+        end
     end)
 
     return ""
 end
 
 local MagicButtonOutcomes = {
+    {
+        func = MagicOutcomePrize,
+        message = "and won [white]%s points![fbc];coins;",
+        weight = 1
+    },
     {
         func = MagicOutcomeBountyAndPrize,
-        message = "and won %s points and also a %s point bounty on themself!",
+        message = "and won [red]%s points[fbc] and also a [red]%s point bounty[fbc] on themself! ;fingers;",
+        weight = 3
+    },
+    {
+        func = MagicOutcomeKleinerFanclub,
+        message = "%s",
+        weight = 5
+    },
+    {
+        func = MagicOutcomeKleinerTeleported,
+        message = "%s",
+        weight = 4,
+    },
+    {
+        func = MagicOutcomeKleinerSlur,
+        message = "%s",
+        weight = 3
+    },
+    {
+        func = MagicOutcomeOverlay,
```

Comment:
this "func/message" thing is confusing, can't you just make the func return the formatted message (or nil if it fails)?

Reply:
i think this is mostly leftover from before i realized some messages might need to be customized. I can go through everything and fix it if you want

Reply:
yeah please do, it's hard to figure out how the message is created and that would help a lot